annotation-file-utilities/src/annotator/find/Criteria.java

```diff
@@ -96,6 +96,14 @@ public boolean isSatisfiedBy(TreePath path) {
    * @return true iff this is the criteria on a receiver
    */
   public boolean isOnReceiver() {
+    ASTPath astPath = getASTPath();
+    if (astPath != null && !astPath.isEmpty()) {
+      ASTPath.ASTEntry entry = astPath.get(-1);
+      return entry.getTreeKind() == Tree.Kind.METHOD
```

Comment:
Is it possible that there is no ASTPathCriterion in this, and yet this contains some criterion whose kind is RECEIVER?  If so, this return statement is incorrect.  The code should return true if the condition is satisfied, and otherwise fall through to the test immediately below (which is currently on line 107).  The same comment applies to other code in this class, which I think should also be changed.